openff/toolkit/topology/molecule.py

```diff
@@ -3028,11 +3031,6 @@ def are_isomorphic(
             [Dict[int,int]] ordered by mol1 indexing {mol1_index: mol2_index}
             If molecules are not isomorphic given input arguments, will return None instead of dict.
         """
-
-        # Do a quick hill formula check first
-        if Molecule.to_hill_formula(mol1) != Molecule.to_hill_formula(mol2):
-            return False, None
-
```

Comment:
This might introduce a performance regression. I do not have a strong intuition as to how often this short-circuit was triggered; I'd think rarely but I don't know.

Reply:
(discussed in our meeting today) Hm, this is a pretty important performance speedup (I think this is used for input validation/conformer handling in `openff-benchmark`), so I'd like to preserve this (either by having a check here for whether the input is a networkx graph or a molecule, or by making something like a private method for `Molecule._object_to_hill_formula(obj)`)